src/FirmwarePlugin/FirmwarePlugin.cc

```diff
@@ -485,7 +485,7 @@ bool FirmwarePlugin::_setFlightModeAndValidate(Vehicle* vehicle, const QString&
         vehicle->setFlightMode(flightMode);
 
         // Wait for vehicle to return flight mode
-        for (int i=0; i<30; i++) {
+        for (int i=0; i<22; i++) {
```

Comment:
This is 2.2 seconds, which is being tried 3 times, so a total of 6.6 seconds. It was 9 seconds before, which is evil because the operator would never expect to still get an action 10 seconds after he operated the UI.

We might want to still lower it to 1.5.